quil-py/quil/instructions/__init__.pyi

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+
+from _calibration import Calibration, MeasureCalibrationDefinition
+from _gate import (
+    Gate as Gate,
+    GateSpecification as GateSpecification,
+    GateModifier as GateModifier,
+    GateDefinition as GateDefinition,
+)
+from _expression import Expression as Expression
+from _qubit import Qubit as Qubit
+
+class Instruction:
+    """
+    A Quil instruction. Each variant corresponds to a possible type of Quil instruction.
+
+    Variants:
+        ``arithmetic``: An arithmetic expression
+        ``calibration_definition``: Corresponds to a `DEFCAL` instruction (not `DEFCAL MEASURE`)
+        ``declaration``: Corresponds to a `DECLARE` statement
+        ``gate``: A Quil quantum gate instruction
+        ``halt``: Corresponds to the `HALT` instruction.
+        ``measure_calibration_definition``: Corresponds to a `DEFCAL MEASURE` instruction.
+        ``nop``: Corresponds to the `NOP` instruction.
+
+    Methods (for each variant):
+        ``is_*``: Returns ``True`` if the instruction is that variant, ``False`` otherwise.
+
+        If the variant has inner data:
```

Comment:
I think this should either be rephrased to indicate what it means to have inner data (i.e. arguments to the instruction) or each variant should indicate whether it has inner data

Reply:
Okay, I expanded on inner data in the docstrings for this and `Expression`